compiler/src/iree/compiler/InputConversion/MHLO/ConvertMHLOToLinalgExt.cpp

```diff
@@ -470,6 +471,140 @@ struct ReverseOpConversion : public OpConversionPattern<mhlo::ReverseOp> {
+        loc, dynSizes, outputValuesType.getShape(),
+        outputValuesType.getElementType());
+    Value initTensorOutputIndices = rewriter.create<mlir::linalg::InitTensorOp>(
+        loc, dynSizes, outputIndicesType.getShape(),
+        outputIndicesType.getElementType());
+    // Check that the last dim size in the output = k
+    if (outputValuesType.getShape().back() != k ||
+        outputIndicesType.getShape().back() != k) {
+      return failure();
+    }
+    // Initialize indices to 0 and values to negative infinity
+    Attribute negInfAttr;
+    if (outputValuesType.getElementType().isa<IntegerType>()) {
+      negInfAttr = rewriter.getIntegerAttr(
+          outputValuesType.getElementType(),
+          APInt::getSignedMinValue(outputValuesType.getElementType()
+                                       .cast<IntegerType>()
+                                       .getWidth()));
+    } else {
+      auto negApFloat = APFloat::getInf(outputValuesType.getElementType()
+                                            .cast<FloatType>()
+                                            .getFloatSemantics(),
+                                        true);
+      negInfAttr =
+          rewriter.getFloatAttr(outputValuesType.getElementType(), negApFloat);
+    }
+    Value negInf = rewriter.create<arith::ConstantOp>(
+        loc, negInfAttr, outputValuesType.getElementType());
+    Value zero = rewriter.create<arith::ConstantOp>(
+        loc, rewriter.getZeroAttr(outputIndicesType.getElementType()));
+    Value negInfTensor =
+        rewriter.create<linalg::FillOp>(loc, negInf, initTensorOutputValues)
+            .getResult(0);
+    Value zeroTensor =
+        rewriter.create<linalg::FillOp>(loc, zero, initTensorOutputIndices)
+            .getResult(0);
+
+    // Replace the CHLO TopK with LinalgExt TopK
+    auto topkOp = rewriter.replaceOpWithNewOp<IREE::LinalgExt::TopkOp>(
+        op, op->getResultTypes(), ValueRange{operand, inputIndices},
+        ValueRange{negInfTensor, zeroTensor}, kDim);
+
+    // Define the region of TopK with a GT comparison
+    SmallVector<Type> types{inputValuesType.getElementType(),
+                            inputValuesType.getElementType()};
+    SmallVector<Location> locations{loc, loc};
+    Block *block = rewriter.createBlock(&topkOp.region(), {}, types, locations);
+    {
+      rewriter.setInsertionPointToStart(&topkOp.region().front());
+      Value lhs = block->getArgument(0);
+      Value rhs = block->getArgument(1);
+      Value condition;
+      if (inputValuesType.getElementType().isa<IntegerType>()) {
+        condition = rewriter.create<arith::CmpIOp>(
+            loc, arith::CmpIPredicate::sge, lhs, rhs);
+      } else {
+        condition = rewriter.create<arith::CmpFOp>(
+            loc, arith::CmpFPredicate::OGT, lhs, rhs);
+      }
+      rewriter.create<IREE::LinalgExt::YieldOp>(loc, condition);
+    }
+
+    rewriter.setInsertionPoint(topkOp);
```

Comment:
We usually use a guard before setting insertion point. When the guard get destroyed, it'll get back to the original insertion point.

```
{
  OpBuilder::InsertionGuard guard(rewriter);
  rewriter.setInsertionPointToStart(block);
   ...
}
```